File: CaseStudies/glass/Implementations/C/src/DerivedValues.c
#include "DerivedValues.h"
#include "Constants.h"

#include <math.h>

void derived_params(InputParameters* params){
  params->asprat = params->a / params->b;
  params->sd = sqrt(pow(params->sdx, 2.0) + pow(params->sdy, 2.0) + pow(params->sdz, 2.0));
  params->ldf = pow((TD / 60.0), (M / 16.0)); 
  params->wtnt = params->w * params->tnt;
  
  if(params->t == 2.50) params->h = 2.16;
  else if(params->t == 2.70) params->h = 2.59;
  else if(params->t == 3.00) params->h = 2.92;
  else if(params->t == 4.00) params->h = 3.78;
  else if(params->t == 5.00) params->h = 4.57;
  else if(params->t == 6.00) params->h = 5.56;
  else if(params->t == 8.00) params->h = 7.42;
  else if(params->t == 10.00) params->h = 9.02;
  else if(params->t == 12.00) params->h = 11.91;
  else if(params->t == 16.00) params->h = 15.09;
  else if(params->t == 19.00) params->h = 18.26;
  else if(params->t == 22.00) params->h = 21.44;
  
  if(params->gt == 1) params->gtf = 1;
  else if(params->gt == 2) params->gtf = 2;
  else if(params->gt == 3) params->gtf = 4;
}

File: code/stable/glassbr/src/java/GlassBR/DerivedValues.java
package GlassBR;

/** \file DerivedValues.java
    \author Nikitha Krithnan and W. Spencer Smith
    \brief Provides the function for calculating derived values
*/
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;

public class DerivedValues {
    
    /** \brief Calculates values that can be immediately derived from the inputs
        \param inParams structure holding the input values
    */
    public static void derived_values(InputParameters inParams) throws IOException {
        PrintWriter outfile;
        outfile = new PrintWriter(new FileWriter(new File("log.txt"), true));
        outfile.println("function derived_values called with inputs: {");
        outfile.print("  inParams = ");
        outfile.println("Instance of InputParameters object");
        outfile.println("  }");
        outfile.close();
        
        inParams.LDF = Math.pow(3.0 / 60.0, 7.0 / 16.0);
        outfile = new PrintWriter(new FileWriter(new File("log.txt"), true));
        outfile.print("var 'inParams.LDF' assigned ");
        outfile.print(inParams.LDF);
        outfile.println(" in module DerivedValues");
        outfile.close();
        
        inParams.SD = Math.sqrt(Math.pow(inParams.SD_x, 2.0) + Math.pow(inParams.SD_y, 2.0) + Math.pow(inParams.SD_z, 2.0));
        outfile = new PrintWriter(new FileWriter(new File("log.txt"), true));
        outfile.print("var 'inParams.SD' assigned ");
        outfile.print(inParams.SD);
        outfile.println(" in module DerivedValues");
        outfile.close();
        
        inParams.w_TNT = inParams.w * inParams.TNT;
        outfile = new PrintWriter(new FileWriter(new File("log.txt"), true));
        outfile.print("var 'inParams.w_TNT' assigned ");
        outfile.print(inParams.w_TNT);
        outfile.println(" in module DerivedValues");
        outfile.close();
    }
}
